packages/runtime/container-runtime/src/dataStore.ts

```diff
@@ -51,7 +51,7 @@ enum AliasState {
     None = "None",
 }
 
-class DataStore implements IDataStore {
+class DataStore implements IDataStore, Partial<{ readonly IFluidHandle: IFluidHandle<FluidObject> | undefined; }> {
```

Comment:
Tony and I discussed this. `Partial<IProvideFluidHandle>` has the issue that the default generic parameter to `IFluidHandle` is `FluidObject & IFluidLoadable`, and the second part makes it complicated to interact with handle in some places because not all things are `IFluidLoadable`. The only way I could work around that was some `as any` usage. In the end, I needed to use `as any` anyway in some places. But Tony mentioned he has *thoughts* on `IFluidLoadable` and might be better to keep this temporary interface free of it.

Reply:
you shouldn't build your own type. do: `Partial<Pick<IFluidHandle<FluidObject>, "IFluidHandle>>`

Reply:
that is equivalent to using IProvideFluidHandle, but supports the generic

Reply:
Doesn't seem to work:

![image](https://user-images.githubusercontent.com/716334/190718692-c8b908aa-506e-41cf-b40f-1bb86e35c539.png)

If I got this right, it's because the `IFluidHandle` property of the `IFluidHandle<T>` interface comes from `IProvideFluidHandle` which doesn't give it an explicit generic type, thus defaulting to `FluidObject & IFluidLoadable`, so the implementation which skips the `IFluidLoadable` part complains just like we saw before.

```typescript
export interface IProvideFluidHandle {
    readonly IFluidHandle: IFluidHandle;
}

export interface IFluidHandle<T = FluidObject & IFluidLoadable> extends IProvideFluidHandle {
  // IFluidHandle property is not defined here
}
```

Any other TS-foo that might be worth trying?

Reply:
will all of this casting just go away if we implement both IFluidHandle and IFluidLoadable? we do need to rationalize those two, but right now they are really coupled

Reply:
i also think having handles that are not loadable could cause problems for users who maybe had the expectation that everything returned by a handle is loadable based on the typing.

Reply:
I'll give that a shot. For my own education, what does "loadable" mean in this context? Anything to do with the loader in the host?

Reply:
it's legacy. loadable things used to participate in the request pattern, but that was removed, so now they just expose handles.